File: app/src/main/java/com/shindra/mapActivity.java
package com.shindra;

import androidx.appcompat.app.AppCompatActivity;

import androidx.fragment.app.FragmentTransaction;
import androidx.lifecycle.ViewModelProvider;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.util.Log;


import com.shindra.arrakis.controls.MapFragment;
import com.shindra.arrakis.controls.Poi;
import com.shindra.arrakis.observable.ObservableExtensionKt;
import com.shindra.arrakis.observable.ObservableListener;
import com.shindra.ctslibrary.apibo.RouteType;
import com.shindra.ctslibrary.bo.Line;
import com.shindra.ctslibrary.bo.Stop;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;

public class mapActivity extends AppCompatActivity {

    private mapFragment frag;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_map);

        getSupportActionBar().setTitle(getString(R.string.ligneText) + getIntent().getStringExtra("LINE_NAME"));

        frag = new mapFragment();
        Bundle bundle = new Bundle();

        getSupportFragmentManager().beginTransaction().add(R.id.map_container,frag).commit();
        frag.setArguments(bundle);
        bundle.putString("lineName",getIntent().getStringExtra("LINE_NAME"));

    }

}

File: app/src/main/java/com/shindra/MapActivity.kt
package com.shindra

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.shindra.Schedule__Fragment.Companion.LINENAME_MESSAGE

class MapActivity : AppCompatActivity() {
    private val logTag = "GpsMapActivity"
    companion object{
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.map_activity)

        val requestedLineName = intent.getStringExtra(Schedule__Fragment.LINENAME_MESSAGE)


        supportActionBar?.title = getString(R.string.Schedule_LineName, requestedLineName)

        //start fragment
        val args = Bundle()
        val fragmentManager = supportFragmentManager
        val fragment = MapFragment()
        val fragmentTransaction = fragmentManager.beginTransaction()

        args.putString(LINENAME_MESSAGE, requestedLineName)
        fragment.arguments = args

        fragmentTransaction.add(R.id.stopfragment, fragment)
        fragmentTransaction.commit()

    }
}